clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -557,6 +558,7 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
      * @param <T> Exception cause type parameter
      * @return The caught exception cause
      */
+    @Timeout(DEFAULT_MAX_WAIT_MS)
```

Comment:
Have you confirmed this will work as expected? That is, if a future is not throwing exception as expected in time, will the operation get terminated?